src/clarity-angular/datagrid/providers/sort.ts

```diff
@@ -18,9 +18,12 @@ export class Sort {
     public get comparator(): Comparator<any> {
         return this._comparator;
     }
+
     public set comparator(value: Comparator<any>) {
-        this._comparator = value;
-        this.emitChange();
+        if (typeof value !== "undefined") {
+            this._comparator = value;
+            this.emitChange();
+        }
```

Comment:
@youdz thanks for the review here. This one was the one I was also apprehensive about so I was waiting until I get your review here 😛 Will fix it now.